handler.go

```diff
@@ -504,10 +449,22 @@ func (s *handler) handle(ctx context.Context, req request, w func(func(io.Writer
 
 			log.Warnf("failed to setup channel in RPC call to '%s': %+v", req.Method, err)
 			stats.Record(ctx, metrics.RPCResponseError.M(1))
-			resp.Error = &respError{
+
+			respErr := &JSONRPCError{
 				Code:    1,
 				Message: err.Error(),
 			}
+
+			if m, ok := err.(RPCErrorCodec); ok {
+				rpcErr, err := m.ToJSONRPCError()
+				if err != nil {
+					log.Warnf("Failed to convert error to JSONRPCError: %v", err)
+				} else {
+					respErr.Data = rpcErr.Data
+				}
+			}
+
+			resp.Error = respErr
```

Comment:
This is unnecessary, we only ever get back an internal error at this point which is why code 1 is being used as the internal error code (custom codes start at 2). You can follow this case down in to websocket.go `handleChanOut`. So all you need to do here is `s/respError/JSONRPCError/` and leave the rest alone.

Reply:
I will remove this from here, wdym by when you say `You can follow this case down in to websocket.go handleChanOut`. I don't see any error in the response there.